test/gtest/ucs/test_profile.cc

```diff
@@ -84,27 +110,93 @@ UCS_PROFILE_FUNC(int, profile_test_func2, (a, b), int a, int b)
     return UCS_PROFILE_CALL(sum, a, b);
 }
 
-const int test_profile::MAX_LINE = __LINE__;
+const int test_profile::MAX_LINE           = __LINE__;
+const unsigned test_profile::NUM_LOCAITONS = 12u;
+
+test_profile::test_profile()
+{
+    pthread_spin_init(&m_tids_lock, 0);
+}
+
+test_profile::~test_profile()
+{
+    pthread_spin_destroy(&m_tids_lock);
+}
+
+void test_profile::add_tid(int tid)
+{
+    pthread_spin_lock(&m_tids_lock);
+    m_tids.insert(tid);
```

Comment:
do we really need `m_tids` and `m_tids_lock`? I see that we only insert to the `set`, but not use `tid`s